components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/impl/BasicAuthenticationHandler.java

```diff
@@ -18,152 +18,136 @@
+            if (splitCredentials.length >= 2) {
                 String userName = splitCredentials[0];
-                String password = splitCredentials[1];
+                char[] password = splitCredentials[1] != null ? splitCredentials[1].toCharArray() : new char[0];
 
-
-                UserStoreManager userStoreManager = null;
                 try {
-                    int tenantId = IdentityTenantUtil.getTenantIdOfUser(userName);
-                    String tenantDomain = MultitenantUtils.getTenantDomain(userName);
-
-                    User user = new User();
-                    user.setUserName(MultitenantUtils.getTenantAwareUsername(userName));
-                    user.setTenantDomain(tenantDomain);
-
-                    authenticationContext.setUser(user);
+                    ImmutablePair<String, String> domainAndUser = decodeTenantDomainAndUserName(userName);
+                    userName = domainAndUser.getLeft();
+                    String domainName = domainAndUser.getRight();
 
+                    IdentityStore identityStore = getRealmService().getIdentityStore();
 
                     //TODO: Related to this https://wso2.org/jira/browse/IDENTITY-4752 - Class IdentityMgtEventListener
                     // : Line 563: Have to check whether why we can't continue
                     //without following lines as previous code.
-                    PrivilegedCarbonContext.startTenantFlow();
-                    PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain);
-                    PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(tenantId);
-
-                    UserRealm userRealm = AuthenticationServiceHolder.getInstance().getRealmService().
-                            getTenantUserRealm(tenantId);
-                    if ( userRealm != null ) {
-                        userStoreManager = (UserStoreManager) userRealm.getUserStoreManager();
-                        boolean isAuthenticated = userStoreManager.authenticate(MultitenantUtils.
-                                getTenantAwareUsername(userName), password);
-                        if ( isAuthenticated ) {
+                    //                    PrivilegedCarbonContext.getCurrentContext()
+
+                    if (identityStore == null) {
+                        String errorMessage =
+                                "Could not get the identity store to autnenticate the user with "
+                                        + "BASIC authentication. Username: "
+                                        + userName;
+                        if (log.isDebugEnabled()) {
+                            log.debug(errorMessage);
+                        }
+                        throw new AuthenticationFailException(errorMessage);
+                    } else {
+
+                        Claim userNameClaim = new Claim("http://wso2.org/claims", "http://wso2.org/claims/username",
```

Comment:
Use the constants, `IdentityMgtConstants.CLAIM_ROOT_DIALECT` and `IdentityMgtConstants.USERNAME_CLAIM`